### PR TITLE
Renenable WebFetchers test

### PR DESCRIPTION
**Pull request by Siedlerchr**
Replace outdated reflection library with [classgraph](https://github.com/classgraph/classgraph) library

Fixes part of #5226 

----

- [ ] Change in CHANGELOG.md described
- [x] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -197,6 +197,8 @@ dependencies {
         exclude module: "log4j-core"
     }
 
+
+    compile "io.github.classgraph:classgraph:4.8.47"
```
Comment (tobiasdiez) [inline build.gradle]: testcompile should be sufficient as we are not using it in production

File: src/main/java/module-info.java [hunk 2]
```diff
@@ -65,4 +65,5 @@
     requires de.saxsys.mvvmfx.validation;
     requires richtextfx;
     requires unirest.java;
+    requires io.github.classgraph;
```
Comment (tobiasdiez) [inline src/main/java/module-info.java]: Move to https://github.com/JabRef/jabref/blob/master/src/test/java/module-info.test

**Comment by Siedlerchr** [comment]
Reflections doesn't work with Java 11 and seems like it's no longer maintained.

**Comment by Siedlerchr** [comment]
Fixed the remaining issues, so I am merging this now.

**Pull request merged**